To: vim_dev@googlegroups.com
Subject: Patch 7.3.141
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.141
Problem:    When a key code is not set get a confusing error message.
Solution:   Change the error message to say the key code is not set.
Files:	    src/option.c, runtime/doc/options.txt


*** ../vim-7.3.140/src/option.c	2011-01-22 00:11:42.000000000 +0100
--- src/option.c	2011-03-22 14:25:52.000000000 +0100
***************
*** 4352,4358 ****
  		    p = find_termcode(key_name);
  		    if (p == NULL)
  		    {
! 			errmsg = (char_u *)N_("E518: Unknown option");
  			goto skip;
  		    }
  		    else
--- 4352,4358 ----
  		    p = find_termcode(key_name);
  		    if (p == NULL)
  		    {
! 			errmsg = (char_u *)N_("E846: Key code not set");
  			goto skip;
  		    }
  		    else
***************
*** 4700,4707 ****
  						|| s[i] == ','
  						|| s[i] == NUL))
  					break;
! 				    /* Count backspaces.  Only a comma with an
! 				     * even number of backspaces before it is
  				     * recognized as a separator */
  				    if (s > origval && s[-1] == '\\')
  					++bs;
--- 4700,4707 ----
  						|| s[i] == ','
  						|| s[i] == NUL))
  					break;
! 				    /* Count backslashes.  Only a comma with an
! 				     * even number of backslashes before it is
  				     * recognized as a separator */
  				    if (s > origval && s[-1] == '\\')
  					++bs;
*** ../vim-7.3.140/runtime/doc/options.txt	2010-12-02 21:43:10.000000000 +0100
--- runtime/doc/options.txt	2011-03-22 14:32:14.000000000 +0100
***************
*** 150,155 ****
--- 150,167 ----
  (the ^[ is a real <Esc> here, use CTRL-V <Esc> to enter it)
  The advantage over a mapping is that it works in all situations.
  
+ You can define any key codes, e.g.: >
+ 	:set t_xy=^[foo;
+ There is no warning for using a name that isn't recognized.  You can map these
+ codes as you like: >
+ 	:map <t_xy> something
+ <								*E846*
+ When a key code is not set, it's like it does not exist.  Trying to get its
+ value will result in an error: >
+ 	:set t_kb=
+ 	:set t_kb
+ 	E846: Key code not set: t_kb
+ 
  The t_xx options cannot be set from a |modeline| or in the |sandbox|, for
  security reasons.
  
*** ../vim-7.3.140/src/version.c	2011-03-22 13:29:20.000000000 +0100
--- src/version.c	2011-03-22 14:32:59.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     141,
  /**/

-- 
"I know that there are people who don't love their fellow man,
and I hate those people!" - Tom Lehrer

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
